test/src/sdk-harness/test_projects/storage_map/src/main.sw

```diff
@@ -0,0 +1,281 @@
+contract;
+
+use std::storage::StorageMap;
+
+pub struct Struct {
+    x: u32,
+    y: b256,
+    z: b256,
+}
+
+pub enum Enum {
+    V1: b256,
+    V2: u64,
+    V3: b256,
+}
+
+storage {
```

Comment:
So we have native types that can be added into storage without the use of the std library. Why can a mapping not be one of those types that is built into ... the compiler? What makes a `u64` able to be put directly into storage but a `mapping` has to be in the library?

Reply:
Well, `StorageMap` is just a `struct` so it's no different than custom types that users can define and declare inside a `storage` block. The compiler handles it exactly as it handles any user-defined type. We can make `StorageMap` a primitive type but I think that just adds bloat to the compiler and it's really not considered good compiler practice. `HashMap` in Rust is a library for example.